### DISCUSSION TITLE
Can't download any video in Edge with licensed add-on

### DISCUSSION THREAD

**Discussion by Rodrigo-Faustini** [aclap-dev/video-downloadhelper#497] [Help and Q&A]
### Your question

Can't download any video in Edge after purchasing license. It shows the download options, but then the error messages below appear. 

- I'm behind a corporate proxy (but logged in, Internet works fine)
- Companion App installed (v2.0.19) 
- Licensed (for Edge) - Is it necessary to purchase a diferent, specific license for each browser? 

{
  "version": "9.0.1.13",
  "target": "microsoft",
  "channel": "stable",
  "page_url": "https://www.youtube.com/watch?v=kF9eVhg9x8g",
  "error": {
    "details": "SideDownload error: : [tcp @ 0000029a5a50c0c0] Connection to tcp://manifest.googlevideo.com:443 failed: Error number -138 occurred\r\n[in#0 @ 0000029a5bccb840] Error opening input: Error number -138 occurred\r\nError opening input file https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1715809153/ei/IddEZuqcN_ja1sQPuIqZ6Ak/ip/201.62.39.32/id/905f5e56183dc7c8/itag/312/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/sgovp/clen%3D481237763%3Bdur%3D932.266%3Bgir%3Dyes%3Bitag%3D299%3Blmt%3D1709616550437415/rqh/1/hls_chunk_host/rr3---sn-jvhhjvg2-cnce.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/Zr/mm/31,29/mn/sn-jvhhjvg2-cnce,sn-gpv7ynel/ms/au,rdu/mv/m/mvi/3/pcm2cms/yes/pl/24/initcwndbps/616250/vprv/1/playlist_type/DVR/dover/13/txp/5535434/mt/1715787176/fvip/5/short_key/1/keepalive/yes/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,sgovp,rqh,xpc,vprv,playlist_type/sig/AJfQdSswRQIgZ5WNA34SnzV0dv4fUWzNfg1R1pWBqqqZYDuNwIo_MEwCIQDF-TJIw7Xx3inLkLOy357w7Xum2YQCbD74S3gfBnFzjA%3D%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pcm2cms,pl,initcwndbps/lsig/AHWaYeowRgIhAJUVZeqI6f78-434AuK_GmUYd2pFWFUdpsq2cUJ6Kjb3AiEA3EYzYKq3ExuOPjVMtOBIp8McHJdDjI_jg70oEJz6Uo8%3D/playlist/index.m3u8.\r\nError opening input files: Error number -138 occurred\r\n",
    "downloadable_id": "downloadable_4370479844616044",
    "error": "coapp_failure",
    "id": "coapp_failure_5ec6ba10-cffe-4459-8074-b8107addec2b"
  }
}

{
  "version": "9.0.1.13",
  "target": "microsoft",
  "channel": "stable",
  "page_url": "https://www.dailymotion.com/video/x8yj646",
  "error": {
    "details": "SideDownload error: : [tcp @ 000001f440b4c040] Connection to tcp://vod.cf.dmcdn.net:443 failed: Error number -138 occurred\r\n[in#0 @ 000001f440b4b1c0] Error opening input: Error number -138 occurred\r\nError opening input file https://vod.cf.dmcdn.net/sec2(mq3txTsZgpW386B0XOQD5jxjUcrsbFIn-m-dChsvPCc5C7w3l6BRaoEp26DlQ0lxYDHoGamHrDT0IxRan8QTjrGvdS8ZJVbHoZt6I60oQIWdK_lm0iwO9V1s54UueAuz7oVPZ6j96IcnGuQu3SXaxtb4hvdjL8cwvTxtVCnB2Wc)/video/247/037/541730742_mp4_h264_aac_fhd_hfr_1.m3u8#cell=cf.\r\nError opening input files: Error number -138 occurred\r\n",
    "downloadable_id": "downloadable_2246994229287341",
    "error": "coapp_failure",
    "id": "coapp_failure_5ce86e60-062c-4b87-86c2-0e6c73f4a162"
  }
}

{
  "version": "9.0.1.13",
  "target": "microsoft",
  "channel": "stable",
  "page_url": "https://twitter.com/home",
  "error": {
    "details": "SideDownload error: : [tcp @ 00000199d27fbb00] Connection to tcp://video.twimg.com:443 failed: Error number -138 occurred\r\n[in#0 @ 00000199d27fb080] Error opening input: Error number -138 occurred\r\nError opening input file https://video.twimg.com/ext_tw_video/1768234880205209600/pu/pl/avc1/944x720/pcW-sIFDPj8sRiOT.m3u8?container=cmaf.\r\nError opening input files: Error number -138 occurred\r\n",
    "downloadable_id": "downloadable_7058359087955871",
    "error": "coapp_failure",
    "id": "coapp_failure_fc7706e0-1d57-4320-a9f5-ee083bc98763"
  }
}

### Addon & CoApp version and details

Version 9.0.1.13 (stable), target microsoft, locale en-US.
Companion App installed (v2.0.19) 
C:\Program Files\DownloadHelper CoApp\vdhcoapp.exe

**Comment by jcfrog** [comment]
hi,

yes you need a license per browser: https://github.com/aclap-dev/video-downloadhelper/wiki#go-premium

Sorry I did not manage to reproduce your issue, I had no problem with your URLs.

<img width="427" alt="image" src="https://github.com/aclap-dev/video-downloadhelper/assets/368367/82c25a4d-fcfa-46ab-89aa-5758e43ea10b">

**Comment by paulrouget** [comment]
> I'm behind a corporate proxy (but logged in, Internet works fine)

Proxy? Not VPN right?

We don't support proxies sadly (only VPN). But that's something we're working on for the next version though.

**Reply by Rodrigo-Faustini** [reply]
Yes, proxy only. Our proxy requires login with username and password. 

In Edge I get those errors, in Firefox it even can't detect downloadable videos ("No media to process in the current tab").

Thank you.

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

My software information is this:

version: 9.0.1.13a1
target: mozilla
channel: beta
lang: en-US
coapp: {"found":true,"path":"C:\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"<removed>"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "M:\\VDH Testing",
    "concurrent_downloads_max": 16,
    "show_success_notification_for_icognito": true,
    "theme": "system",
    "view_options": {
        "all_tabs": true,
        "low_quality": true,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": true,
        "hide_downloaded": false
    },
    "default_action": "download_as",
    "use_wide_ui": true,
    "use_sidebar": false,
    "open_count_store": 1
}

Although I usually prefer to report it like this:

Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.13a1 beta, CoApp 2.0.19

I tested your videos, too.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/07401eaf-b0dc-42f0-9a39-fc638ce7ae20)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/ecc5f5a0-6ab8-439d-9cae-d3517f76e0bb)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/4b2fa734-b62e-4e86-8de3-9b1d7a2e6fc3)
![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/ddfec823-078f-4e2d-9b4f-e7990504cce2)

So there was no problem with your YouTube video.

For the Dailymotion video . . .

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/fd6cb98d-ca71-4a28-b06b-27ab7c20b6c3)
![#06](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/13d1421f-1ac9-4fca-8aab-7af8e4d98b9e)

The problem report l got is this:

{
  "version": "9.0.1.13a1",
  "target": "mozilla",
  "channel": "beta",
  "page_url": "https://www.dailymotion.com/video/x8yj646",
  "error": {
    "error": "coapp_failure",
    "details": "SideDownload error: : [tcp @ 00000236f2860e40] Connection to tcp://dmxleo.dailymotion.com:443 failed: Error number -138 occurred\r\n[in#0 @ 00000236f285b640] Error opening input: Error number -138 occurred\r\nError opening input file https://public-prod-dspcookiematching.dmxleo.com/dmx?target=https%3A%2F%2Fdmxleo.dailymotion.com/cdn/manifest/video/x8yj646.m3u8%3Fauth%3Dx%26cs%3Ddmx_[CSDATA].\r\nError opening input files: Error number -138 occurred\r\n",
    "id": "coapp_failure_24f72ec5-ed1b-45bb-b7b1-c0f8a6c25781",
    "downloadable_id": "downloadable_1594349371449200"
  }
}

I tried each of the 3 offered variants & they all got this error.

So I went to Plan B.  There was nothing usable in the Network Monitor.  So I clicked the play button in the player on the web page.  First, some ad played.  I couldn't avoid it.  Then the player stalled, as you can see in the following image.

![#07](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/d7ad57da-2b1c-4da3-a060-7a3e5c389141)

But as you can see, that did give me something usable in the Network Monitor.

Running the URL of the HLS master manifest highlighted in that image through ffprobe got me these results:
[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/15325735/ffprobe.txt)

Using that information, I formulated an invocation of ffmpeg & ran that.

![#08](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/85d77c74-df23-4708-bbe9-88463c461d89)

This is the log of the ffmpeg run:
[Dailymotion mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/15325744/Dailymotion.mp4.Log.txt)

Here are the results retrieved by ffmpeg:

![#09](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/3e635dd1-6444-4cf4-99d4-7525a2cb6fb8)
![#10](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/7acdd6b3-61f8-4a07-9eea-0d7832afef0c)

And then I realized I'm quite dumb.  If playing the video in the web page finally gave a usable HLS master manifest, surely VDH would have recognized it.

![#11](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/c66ff988-02aa-4112-8da2-e75b936c0ccb)
![#12](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/dfeb70d4-8dae-469f-8bf8-07247c21999b)
![#13](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/60df3a9e-bd32-402e-8686-7a78e0b5f315)

So VDH downloaded this one as well, although it required sitting through an ad.  Curious that the page never did manage to actually launch the video, but VDH downloaded it anyway.

Now, in retrospect, I'm guessing that the length 0 things VDH failed to download were the ads.  Gee, isn't that too bad.

I suppose my results confirm @paulrouget Paul's comment that proxies are not supported.  Both @jcfrog Jérôme & I are not using your proxy, obviously.  I'm not using a proxy of any kind.  I assume Jérôme isn't either.  So that would seem to be your issue.  I guess you just have to do your downloads at home instead of at work.  I would agree with you that that is quite annoying.

**Reply by Rodrigo-Faustini** [reply]
Wow, this is an almost forensic, CSI analysis - very well done! Thank you so much for your effort! 

Since both Youtube and Dailymotion websites are not blocked by our proxy, and also considering that I'm using Youtube Premium (which doesn't show ads), I didn't expect VDH would not work through our proxy. However, there's the application side of the story. 

And yes, the lack of proxy support is quite annoying because sometimes I need to save videos to embed in offline work presentations. So, I'd really like to see proxy support added in a near future version.

Thank you.

**Reply by GrampaWildWilly** [reply]
I've never used a proxy.  I can't say I know much about them beyond the word.  Isn't there a way to temporarily get around the proxy?  Can't you disable it long enough to do your downloads?

As for forensics, what I did is pretty typical of what I always do when VDH doesn't work.  Except I was too slow on the uptake to realize VDH in fact did work on Dailymotion, so my Plan B was actually completely unnecessary.  But that Plan B is something I have done countless times.  You can find it scattered about in this forum, & even more times back on our now-defunct former Google forum.

**Reply by Rodrigo-Faustini** [reply]
Unfortunately we can't disable the proxy service. We could try to add an exception rule to it (that's what we do for applications that don't work behind proxies), but I'm pretty sure it won't work because such workaround just allow the access to the domain or IP# of a target web host, instead of allowing an specific application to bypass the proxy at all. The best option would be if the application supported communicating through proxies.

Thank you.
